src/main/daml/Daml/Finance/Interface/Instrument/Swap/InterestRate/Types.daml

```diff
@@ -0,0 +1,37 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Interface.Instrument.Swap.InterestRate.Types where
+
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (K)
+import Daml.Finance.Interface.Types.Common (InstrumentKey(..), PartiesMap)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.Schedule (PeriodicSchedule(..))
+
+data InterestRate = InterestRate
+  with
+    instrumentKey : InstrumentKey
+      -- ^ The instrument's key.
+    description : Text
+      -- ^ The description of the swap.
+    referenceRateId : Text
+      -- ^ The floating rate reference ID. For example, in case of "3M Euribor vs 2.5% fix" this should be a valid reference to the "3M Euribor" reference rate.
+    issuerPaysFix : Bool
+      -- ^ Indicate whether the issuer of the instrument pays the fix or the floating leg of the swap.
+    fixRate : Decimal
+      -- ^ The interest rate of the fix leg. For example, in case of "3M Euribor vs 2.5% fix" this should be 0.025.
+    periodicSchedule : PeriodicSchedule
+      -- ^ The schedule for the periodic swap payments.
+    holidayCalendarIds : [Text]
+      -- ^ The identifier of the holiday calendar to be used for the swap payment schedule.
+    calendarDataProvider : Party
+      -- ^ The reference data provider to use for the holiday calendar.
+    dayCountConvention : DayCountConventionEnum
+      -- ^ The day count convention used to calculate day count fractions. For example: Act360.
+    currency : BaseInstrument.K
+      -- ^ The currency of the swap. For example, if the swap pays in USD this should be a USD cash instrument.
+    lastEventTimestamp : Time
+      -- ^ (Market) time of the last recorded lifecycle event. If no event has occurred yet, the time of creation should be used.
+    observers : PartiesMap
+      -- ^ The instrument's observers.
```

Comment:
@GeorgSchneider, I've left observers in the type definition which means it can be viewed from the relative interface (ie., in this case the `InterestRate.Interface`). I can remove it if you feel its not necessary in the view.